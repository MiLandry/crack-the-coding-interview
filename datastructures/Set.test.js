const MySet = require('./Set');

let set

beforeEach(() => {

set = new MySet()
});

test('it adds', () => {
  set.add('a')
  expect(set)
    .toMatchSnapshot()

})

test('it adds 2', () => {
  set.add('a')
  set.add('a')
  expect(set)
    .toMatchSnapshot()

})

test('it adds 2 different', () => {
  set.add('a')
  set.add('b')
  expect(set)
    .toMatchSnapshot()

})

test('has', () => {
  set.add('a')
  set.add('b')
  expect(set.has('a'))
  .toBeTruthy()
})

test('has', () => {
  set.add('a')
  set.add('b')
  expect(set.has('z'))
  .toBeFalsy()
})

test('remove', () => {
  set.add('a')
  set.add('b')
  set.remove('a')
  expect(set)
  .toMatchSnapshot()
})

