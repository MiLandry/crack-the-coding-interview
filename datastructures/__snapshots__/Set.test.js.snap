// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it adds 1`] = `
MySet {
  "data": Object {
    "a": true,
  },
}
`;

exports[`it adds 2 1`] = `
MySet {
  "data": Object {
    "a": true,
  },
}
`;

exports[`it adds 2 different 1`] = `
MySet {
  "data": Object {
    "a": true,
    "b": true,
  },
}
`;

exports[`remove 1`] = `
MySet {
  "data": Object {
    "b": true,
  },
}
`;
